envoy/registry/registry.h

```diff
@@ -618,6 +618,16 @@ template <class T, class Base> class RegisterInternalFactory {
   static Envoy::Registry::RegisterFactory</* NOLINT(fuchsia-statically-constructed-objects) */     \
                                           FACTORY, BASE>                                           \
       FACTORY##_registered
+/**
+ * Macro used for static registration with deprecated name.
+ */
+#define REGISTER_FACTORY_D(FACTORY, BASE, DEPRECATED_NAME)                                         \
```

Comment:
I'd lean towards a readable macro name, like LEGACY_REGISTER_FACTORY.
Alternately and maybe as a follow-up, can we get rid of these yet?  cc @mattklein123 
/wait